### PR TITLE
[SPARK-32251][SQL][TESTS][FOLLOWUP] improve SQL keyword test

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Improve the `SQLKeywordSuite` so that:
1. it checks keywords under default mode as well
2. it checks if there are typos in the doc (found one and fixed in this PR)

### Why are the changes needed?

better test coverage

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @maropu

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/SQLKeywordSuite.scala [hunk 1]
```diff
@@ -146,11 +159,32 @@ class SQLKeywordSuite extends SparkFunSuite with SQLKeywordUtils {
...
+    val nonReservedInDoc = keywordsInDoc.filter(_.apply(1) == "non-reserved").map(_.head).toSet
+    if (nonReservedKeywordsInAnsiMode != nonReservedInDoc) {
+      val misImplemented = ((nonReservedInDoc -- nonReservedKeywordsInAnsiMode) ++
+        (nonReservedKeywordsInAnsiMode -- nonReservedInDoc)).toSeq.sorted
+      fail("Some keywords are documented and implemented inconsistently: " +
+        misImplemented.mkString(", "))
+    }
+  }
+
+  test("Spark keywords are documented correctly under default mode") {
```
Comment (maropu) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/SQLKeywordSuite.scala:177]: Oh, looks nice!

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/SQLKeywordSuite.scala [hunk 2]
```diff
@@ -146,11 +159,32 @@ class SQLKeywordSuite extends SparkFunSuite with SQLKeywordUtils {
     }
   }
 
-  test("Spark keywords are documented correctly") {
-    val reservedKeywordsInDoc = keywordsInDoc.filter(_.apply(1) == "reserved").map(_.head).toSet
-    if (reservedKeywordsInAnsiMode != reservedKeywordsInDoc) {
-      val misImplemented = (reservedKeywordsInDoc -- reservedKeywordsInAnsiMode).toSeq.sorted
-      fail("Some keywords are documented as reserved but actually not: " +
+  test("Spark keywords are documented correctly under ansi mode") {
```
Comment (maropu) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/SQLKeywordSuite.scala:162]: nit: `ansi` -> `ANSI`; the uppercase looks better?

**Review by maropu — APPROVED** [review]
LGTM if the tests pass.

**Comment by cloud-fan** [comment]
the last commit just update test name, merging to master/3.0, thanks for the review!

**Pull request closed without merging**